src/solvers/dg/2d/dg.jl

```diff
@@ -1579,43 +1552,42 @@ function prolong2mortars!(dg::Dg2D, ::Val{:l2}, thread_cache)
     end
 
     # Interpolate large element face data to small interface locations
-    for v in 1:nvariables(dg)
-      if dg.l2mortars.large_sides[m] == 1 # -> large element on left side
-        if dg.l2mortars.orientations[m] == 1
-          # L2 mortars in x-direction
-          for l in 1:nnodes(dg)
-            u_large[v, l] = dg.elements.u[v, nnodes(dg), l, large_element_id]
-          end
-        else
-          # L2 mortars in y-direction
-          for l in 1:nnodes(dg)
-            u_large[v, l] = dg.elements.u[v, l, nnodes(dg), large_element_id]
-          end
-        end
-        @views dg.l2mortars.u_upper[1, v, :, m] .= dg.mortar_forward_upper * u_large[v, :]
-        @views dg.l2mortars.u_lower[1, v, :, m] .= dg.mortar_forward_lower * u_large[v, :]
-      else # large_sides[m] == 2 -> large element on right side
-        if dg.l2mortars.orientations[m] == 1
-          # L2 mortars in x-direction
-          for l in 1:nnodes(dg)
-            u_large[v, l] = dg.elements.u[v, 1, l, large_element_id]
-          end
-        else
-          # L2 mortars in y-direction
-          for l in 1:nnodes(dg)
-            u_large[v, l] = dg.elements.u[v, l, 1, large_element_id]
-          end
-        end
-        @views dg.l2mortars.u_upper[2, v, :, m] .= dg.mortar_forward_upper * u_large[v, :]
-        @views dg.l2mortars.u_lower[2, v, :, m] .= dg.mortar_forward_lower * u_large[v, :]
+    if dg.l2mortars.large_sides[m] == 1 # -> large element on left side
+      leftright = 1
+      if dg.l2mortars.orientations[m] == 1
+        # L2 mortars in x-direction
+        u_large = view(dg.elements.u, :, nnodes(dg), :, large_element_id)
+        interpolate_mortar_values!(dg, mortar_type, leftright, m, u_large)
+      else
+        # L2 mortars in y-direction
+        u_large = view(dg.elements.u, :, :, nnodes(dg), large_element_id)
+        interpolate_mortar_values!(dg, mortar_type, leftright, m, u_large)
+      end
+    else # large_sides[m] == 2 -> large element on right side
+      leftright = 2
+      if dg.l2mortars.orientations[m] == 1
+        # L2 mortars in x-direction
+        u_large = view(dg.elements.u, :, 1, :, large_element_id)
+        interpolate_mortar_values!(dg, mortar_type, leftright, m, u_large)
+      else
+        # L2 mortars in y-direction
+        u_large = view(dg.elements.u, :, :, 1, large_element_id)
+        interpolate_mortar_values!(dg, mortar_type, leftright, m, u_large)
       end
     end
   end
 end
 
+@inline function interpolate_mortar_values!(dg::Dg2D, ::Val{:l2}, leftright, m,
+                                            u_large)
+  multiply_dimensionwise!(view(dg.l2mortars.u_upper, leftright, :, :, m), dg.mortar_forward_upper, u_large)
+  multiply_dimensionwise!(view(dg.l2mortars.u_lower, leftright, :, :, m), dg.mortar_forward_lower, u_large)
+  return nothing
+end
+
```

Comment:
This method needs a comment or - better yet - a docstring that explains its purpose and what it does.

Reply:
:+1: